programs/candidate_staking/src/lib.rs

```diff
@@ -16,6 +26,31 @@ pub mod candidate_staking {
 
         Ok(())
     }
+
+    pub fn stake(ctx: Context<Stake>, job_ad_id: String, base_bump: u8, general_bump: u8, application_bump: u8, job_bump: u8, amount: u32) -> Result<()> {
+
+        let general_parameter = &mut ctx.accounts.general_account;
+        let job_parameter = &mut ctx.accounts.job_account;
+        let application_parameter= &mut ctx.accounts.application_account;
+
+        if general_parameter.mint == ctx.accounts.token_mint.key() {
+            msg!("Mint is matching");
+
+            let already_staked_amount = application_parameter.stake_amount;
+            let max_amount = job_parameter.max_amount_per_application;
+            if already_staked_amount + amount < max_amount {
+                msg!("You can transfer");
+            }
+            else{
+                msg!("You cannot transfer");
+            }
+        }
+        else{
+            msg!("Mint is not matching");
```

Comment:
Why not return errors in case of not matching? but again, if you do the check in the macro it will be done automatically and this instruction will only run when the check succeeds

Reply:
i will return errors eventually, this is just for testing